### PR TITLE
fix(CX-3274): Replace deprecated fields in Me and Collector Profile

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves [CX-3274] 

### Description

This PR is a follow-up on artsy/metaphysics#4655

This PR replaces the deprecated fields `emailConfirmed` and `identityVerified` with  `isEmailConfirmed` and `isIdentityVerified` respectively

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Replace emailConfirmed and identityVerified with isEmailConfirmed and isIdentityVerified - mrsltun

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3274]: https://artsyproduct.atlassian.net/browse/CX-3274?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by olerichter00 — APPROVED** [review]
🧹

**Review by MounirDhahri — APPROVED** [review]

File: ios/Artsy/Models/API_Models/Social/User.m [hunk 1]
```diff
@@ -24,7 +24,7 @@ + (NSDictionary *)JSONKeyPathsByPropertyKey
         ar_keypath(User.new, userID) : @"id",
         ar_keypath(User.new, email) : @"email",
         ar_keypath(User.new, defaultProfileID) : @"default_profile_id",
-        ar_keypath(User.new, identityVerified) : @"identity_verified",
+        ar_keypath(User.new, isIdentityVerified) : @"identity_verified",
```
[the remark this answers is not in the document]
Comment (MrSltun) [reply ios/Artsy/Models/API_Models/Social/User.m:27]: Not sure that we are querying from the native side, but when I saw `identity_verified` (the variable name in Gravity) I assumed that we do and updated it

[the remark this answers is not in the document]
Comment (MounirDhahri) [reply ios/Artsy/Models/API_Models/Social/User.m:27]: probably we don't need to update this then

[the remark this answers is not in the document]
Comment (MrSltun) [reply ios/Artsy/Models/API_Models/Social/User.m:27]: Okay! I'll update the code now

**Pull request merged**